server/stream.go

```diff
@@ -4194,6 +4206,30 @@ func getExpectedLastSeqPerSubjectForSubject(hdr []byte) string {
 	return string(getHeader(JSExpectedLastSubjSeqSubj, hdr))
 }
 
+// Fast lookup of the message TTL:
+// - Positive return value: duration in seconds.
+// - Zero return value: no TTL or parse error.
+// - Negative return value: don't expire.
+func getMessageTTL(hdr []byte) (int64, error) {
+	ttl := getHeader(JSMessageTTL, hdr)
+	if len(ttl) == 0 {
+		return 0, nil
+	}
+	sttl := bytesToString(ttl)
+	dur, err := time.ParseDuration(sttl)
```

Comment:
I thought we were going to use the one we have for configs since it is expanded past what std lib can parse and may be applicable.

Reply:
The `parseDuration` for the config has a function signature that only makes sense for the config parser and returning errors/warnings etc. It also turns out that the `parseDuration` helper is just falling `time.ParseDuration` too.

Reply:
ok I thought it was wider.